### ISSUE TITLE
Resource checks initiated through the web UI falsely appear to resolve

### ISSUE THREAD

**Issue by jamieklassen** [concourse/concourse#3863]
## Bug Report

When resource checking takes more than five seconds, the UI (misleadingly) says "checking successfully" even when the current check has not resolved.

## Steps to Reproduce

1. Visit a resource page
1. Click the refresh icon to check
1. Wait 5 seconds

## Expected Results

You continue to see a spinner and the message "currently checking"

## Actual Results

You see a green check (or whatever the most recent status in the database is) and the corresponding status message.

## Additional Context

The "last checked" timestamp remains accurate, so this is definitely a UI bug.

## Version Info

* Concourse version: 5.2 (master)
* Deployment type (BOSH/Docker/binary): BOSH
* Infrastructure/IaaS: GCP (prod)
* Browser (if applicable): chrome (N/A?)
* Did this used to work? I don't think so.

**Comment by pivotal-jamie-klassen** [comment]
@kcmannem made a nice [GIF](https://imgur.com/e5a77H0) of this in the case when the most recent status in the database is 'failing to check'.

**Comment by pivotal-jamie-klassen** [comment]
I don't know if the upcoming ephemeral check containers work will render this impossible, but it might be nice for the resources API to tell you whether a check is currently in progress, so that when you load the resource page you could see 'currently checking' and not be tempted to spam the check button. @ddadlani @pivotal-jwinters

**Comment by pivotal-jwinters** [comment]
@pivotal-jamie-klassen this is definitely doable (even with ephemeral check containers). @xtremerui and I have been talking about this, and it should be quite easy to implement. Not sure if it will make it into #3788 though.

**Comment by kcmannem** [comment]
yeah I think it should still be possible with ephemeral containers
